src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -44,6 +44,13 @@ static const wchar_t* ShaderStageToString(asset::IShader::E_SHADER_STAGE stage)
         return L"as";
     case asset::IShader::E_SHADER_STAGE::ESS_MESH:
         return L"ms";
+    case asset::IShader::E_SHADER_STAGE::ESS_RAYGEN: [[fallthrough]];
+    case asset::IShader::E_SHADER_STAGE::ESS_ANY_HIT: [[fallthrough]];
+    case asset::IShader::E_SHADER_STAGE::ESS_CLOSEST_HIT: [[fallthrough]];
+    case asset::IShader::E_SHADER_STAGE::ESS_MISS: [[fallthrough]];
+    case asset::IShader::E_SHADER_STAGE::ESS_INTERSECTION: [[fallthrough]];
+    case asset::IShader::E_SHADER_STAGE::ESS_CALLABLE: 
+        return L"lib";
```

Comment:
@alichraghi what were the old problems when making a `lib` stage shader with our HLSL stuff?

I remember we only had issues with the Compute and Graphics pipelines using it

Reply:
ok lets reheat your PR with the lib target and test it on an example that really needs unscrewing of multiple preprocessor and template compilations (maybe arithmetic unit test?)